### PR TITLE
Add distributed locking to Check Enforcer

### PR DESCRIPTION
**Pull request by mitchdenny**
This PR adds distributed locking (via Azure Storage blob leases) to serialize operations against GitHub. This is a MVP of this approach. If this works and we don't have any further issues with GitHub getting into an inconsistent state then we can look at optimizing this more.

### CODE REVIEW

File: tools/check-enforcer/Azure.Sdk.Tools.CheckEnforcer/Azure.Sdk.Tools.CheckEnforcer.csproj [hunk 1]
```diff
@@ -7,6 +7,7 @@
     <PackageReference Include="Azure.Identity" Version="1.0.0-preview.4" />
     <PackageReference Include="Azure.Security.KeyVault.Keys" Version="4.0.0-preview.4" />
     <PackageReference Include="Azure.Security.KeyVault.Secrets" Version="4.0.0-preview.4" />
+    <PackageReference Include="Azure.Storage.Blobs" Version="12.0.0-preview.3" />
```
Comment (weshaggard) [inline tools/check-enforcer/Azure.Sdk.Tools.CheckEnforcer/Azure.Sdk.Tools.CheckEnforcer.csproj]: did you not want to use preview.4 for storage like the others?

Comment (mitchdenny) [reply tools/check-enforcer/Azure.Sdk.Tools.CheckEnforcer/Azure.Sdk.Tools.CheckEnforcer.csproj]: Updated - thanks - I had meant to do that.

Comment (mitchdenny) [reply tools/check-enforcer/Azure.Sdk.Tools.CheckEnforcer/Azure.Sdk.Tools.CheckEnforcer.csproj]: Ended up reverting that change due to a compat issue. The functions team is aware.

**Comment by mitchdenny** [comment]
@weshaggard this is ready for a review before I merge. Lots of logging code which I'd like to retain since it makes debugging easier for when I come back to it.

**Comment by mitchdenny** [comment]
@weshaggard planning on merging this soon since this represents what is running in production.

**Pull request merged**